lib/api/src/ticketAPI.cpp

```diff
@@ -83,336 +94,587 @@ namespace irods::administration::ticket
                        std::string_view commandModifier4)
     {
         return ticket_manager(conn,
                               command,
-                              ticket_IDentifier,
-                              commandModifier,
+                              ticket_Identifier,
+                              commandModifier1,
                               commandModifier2,
                               commandModifier3,
                               commandModifier4,
                               false);
     }
 
-    int create_ticket(RxComm& conn, ticket_type _type, std::string_view obj_path, std::string_view ticket_name)
+    void create_ticket(RxComm& conn, ticket_type _type, std::string_view obj_path, std::string_view ticket_name)
     {
-        if (_type == ticket_type::READ) {
-            return create_read_ticket(conn, obj_path, ticket_name);
+        if (_type == ticket_type::read) {
+            ticket_manager(conn, "create", ticket_name, "read", obj_path, ticket_name, "");
         }
-        else if (_type == ticket_type::WRITE) {
-            return create_write_ticket(conn, obj_path, ticket_name);
-        }
-        return 1; // Ticket type not defined
-    }
-    int create_ticket(RxComm& conn, ticket_type _type, std::string_view obj_path)
-    {
-        if (_type == ticket_type::READ) {
-            return create_read_ticket(conn, obj_path);
+        else if (_type == ticket_type::write) {
+            ticket_manager(conn, "create", ticket_name, "write", obj_path, ticket_name, "");
         }
-        else if (_type == ticket_type::WRITE) {
-            return create_write_ticket(conn, obj_path);
+        else {
+            throw std::invalid_argument("Ticket type is not defined");
         }
-        return 1; // Ticket type not defined
     }
-
-    int create_read_ticket(RxComm& conn, std::string_view obj_path, std::string_view ticket_name)
-    {
-        return ticket_manager(conn, "create", ticket_name, "read", obj_path, ticket_name, "");
-    }
-    int create_read_ticket(RxComm& conn, std::string_view obj_path)
+    std::string_view create_ticket(RxComm& conn, ticket_type _type, std::string_view obj_path)
     {
         char myTicket[30];
         make_ticket_name(myTicket);
 
-        return ticket_manager(conn, "create", myTicket, "read", obj_path, myTicket, "");
+        if (_type == ticket_type::read) {
+            ticket_manager(conn, "create", myTicket, "read", obj_path, myTicket, "");
+            return myTicket;
+        }
+        else if (_type == ticket_type::write) {
+            ticket_manager(conn, "create", myTicket, "write", obj_path, myTicket, "");
+            return myTicket;
+        }
+        else {
+            throw std::invalid_argument("Ticket type is not defined");
+        }
     }
 
-    int create_write_ticket(RxComm& conn, std::string_view obj_path, std::string_view ticket_name)
+    void create_ticket(admin_tag,
```

Comment:
All the comments on the non-admin versions of these functions apply to the admin versions as well.

Reply:
@secretstf ping